File: security_login_basic_thymeleaf/src/main/java/com/login/controller/AccountController.java
package com.login.controller;

import com.login.form.LoginForm;
import com.login.repository.UserRepository;
import com.login.service.AccountService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Controller;
import org.springframework.validation.Errors;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;

import javax.validation.Valid;

@Controller
@RequestMapping("/account")
@RequiredArgsConstructor
public class AccountController {
    private final AccountService accountService;

//    @PostMapping("/login")
//    public String signUpSubmit(@Valid LoginForm loginForm, Errors errors) {
//        /* error 발생시 다시 화면으로 돌아간다 */
//        if (errors.hasErrors()) {
//            return "login";
//        }
//
//        /* 로그인 */
//        accountService.login(loginForm);
//
//        /* 로그인 완료시, 메인페이지로 이동 */
//        return "redirect:/";
//    }
}

File: src/main/kotlin/com/studyolle/interfaces/AccountController.kt
package com.studyolle.interfaces

import com.studyolle.application.AccountFacade
import com.studyolle.interfaces.dto.AccountDto
import com.studyolle.interfaces.mapper.AccountDtoMapper
import lombok.RequiredArgsConstructor
import lombok.extern.slf4j.Slf4j
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import javax.validation.Valid

@RestController
@RequiredArgsConstructor
@Slf4j
@RequestMapping("/account")
class AccountController(
    private val accountFacade: AccountFacade,
) {
    @PostMapping
    fun signUpSubmit(
        @RequestBody @Valid signUpForm: AccountDto.SignUpForm
    ) {
        accountFacade.processNewAccount(AccountDtoMapper.of(signUpForm))
    }
}